# Task 2: Thomas Fire Perimeter

### EDS 220 Homework 4

Kimberlee Wong (https://github.com/kimberleewong/eds220-hwk4)

## About

### Purpose
This notebook shows the initial data exploration of a dataframe of California fire perimeters. The data wrangling steps that were taken in order to filter for the specific Thomas Fire in 2017 are also included. By doing this, I was able to create a new file that contains only the needed region, which will be used in the next steps of the exercise in another notebook. 

### Highlights
- I think the data import is an important aspect to highlight because there is a difference in importing geodata than regular data, so this notebook shows how to do it.
- 

### About the Data

The shapefile used in this exercise comes from the data catalog owned by the United States government. It is specific to the state of California and public, so it was intended for public access and use. Their metadata states that this data provides a "a reasonable view of the spatial distribution of past large fires but is in no way complete." But, for our uses, it has the adequate amount of information.

### References

Data.gov Data Catalogue, California Fire Perimeters (all) [Data file] Available from: https://catalog.data.gov/dataset/california-fire-perimeters-all-b3436. Access date: November 23, 2024.

## 1. Import Libraries

In [1]:
import pandas as pd
import geopandas as gpd
import os
import matplotlib.pyplot as plt

## 2. Import Data

In [7]:
# Use a filepath to import geodata
california_fire_peri = gpd.read_file(os.path.join('initial_data', 'California_Fire_Perimeters_1379327890478655659', 'California_Fire_Perimeters_(all).shp'))

## 3. Explore Data

In [6]:
print(f"CRS: {california_fire_peri.crs}")

print('Is this CRS projected? ', california_fire_peri.crs.is_projected)

print('Is this CRS projected? ', california_fire_peri.crs.is_geographic)

CRS: epsg:3857
Is this CRS projected?  True
Is this CRS projected?  False


In [4]:
california_fire_peri.head(3)

YEAR_ STATE AGENCY UNIT_ID  FIRE_NAME   INC_NUM  ALARM_DATE   CONT_DATE  \
0   2023    CA    CDF     SKU  WHITWORTH  00004808  2023-06-17  2023-06-17   
1   2023    CA    LRA     BTU     KAISER  00010225  2023-06-02  2023-06-02   
2   2023    CA    CDF     AEU    JACKSON  00017640  2023-07-01  2023-07-02   

   CAUSE  C_METHOD  OBJECTIVE  GIS_ACRES COMMENTS COMPLEX_NA  \
0      5         1          1    5.72913     None       None   
1      5         1          1   13.60240     None       None   
2      2         1          1   27.81450     None       None   

                                  IRWINID FIRE_NUM COMPLEX_ID  DECADES  \
0  {7985848C-0AC2-4BA4-8F0E-29F778652E61}     None       None     2020   
1  {43EBCC88-B3AC-48EB-8EF5-417FE0939CCF}     None       None     2020   
2  {B64E1355-BF1D-441A-95D0-BC1FBB93483B}     None       None     2020   

                                            geometry  
0  POLYGON ((-13682443.000 5091132.739, -13682445...  
1  POLYGON ((-13576727.142 4841226.161, -13576726...  
2  POLYGON ((-13459243.000 4621236.000, -13458968...

In my initial exploration, I learned that the California Fires Perimeters geodata is set in CRS EPSG 3857 and projected, not geographic. In addition to this, I checked out the top three rows of the dataframe. By doing this, I can see that the names of the fires and the year they occurred are included, so it should be straightforward to filter to just the 2017 Thomas Fire. 